maintenance_request_purchase/models/maintenance_request.py

```diff
@@ -23,6 +23,41 @@ class MaintenanceRequest(models.Model):
         groups="purchase.group_purchase_user",
     )
 
+    total_purchase_amount = fields.Monetary(
+        compute="_compute_total_purchase_amount",
+        store=True,
+        groups="purchase.group_purchase_user",
+        currency_field="currency_id",
+    )
+
+    currency_id = fields.Many2one(
+        related="company_id.currency_id",
+        store=True,
+        readonly=True,
+    )
+
+    @api.depends(
+        "purchase_order_ids.amount_total",
+        "purchase_order_ids.currency_id",
```

Comment:
Missing new dependency:
```suggestion
        "purchase_order_ids.state",
```